internal/telemetry/prometheus.go

```diff
@@ -27,6 +31,10 @@ func getKey(metric string, labels map[string]string) string {
 
 func NewCounter(metric string, labels map[string]string) prometheus.Counter {
 	metricKey := getKey(metric, labels)
+
+	counterMetricMutex.Lock()
+	defer counterMetricMutex.Unlock()
+
 	if _, ok := counterMetricMap[metricKey]; !ok {
 		counterMetricMap[metricKey] = promauto.NewCounter(prometheus.CounterOpts{Name: metric, ConstLabels: labels})
```

Comment:
Can we also simplify the logic here ? We are doing search 2 times to return a value.

```
metric, ok := counterMetricMap[metricKey]
if ok {
        return metric
}
newMetric := promauto.NewCounter(prometheus.CounterOpts{Name: metric, ConstLabels: labels})
counterMetricMap[metricKey] = newMetric
return newMetric
```

Reply:
sure